### DISCUSSION TITLE
How to get the train loader of a single specific domain?

### DISCUSSION THREAD

**Discussion by txsing** [p-lambda/wilds#123] [Q&A] [answered]
For example, for iWildCam dataset, I want to write a training data loader of the specific location No. 296
Are there any code snippets for that?

**Comment by ssagawa** [comment, marked as answer]
You should be able to select a relevant subset using `WILDSSubset` (defined [here](https://github.com/p-lambda/wilds/blob/f384c21c67ee58ab527d8868f6197e67c24764d4/wilds/datasets/wilds_dataset.py#L473)) and then pass the subset into the usual `get_train_loader` function.

**Reply by txsing** [reply]
Thanks a lot for the reply,  but still I did not find clues from the WILDSSubset code comments on how to get a subset of a specific domain from the full iwildcame dataset.  Would u mind giving some pseudocode for that?

A lot of thanks ahead.

**Reply by ssagawa** [reply]
You can do something like this:
```
iwc_dataset = get_dataset(dataset="iwildcam")
idx, = np.where(np.logical_and(iwc_dataset.metadata_array[:, iwc_dataset.metadata_fields.index('location')].numpy()==LOCATION,
                               iwc_dataset.split_array==TRAIN))
subset = WILDSSubset(iwc_dataset, idx, transform)
```
